tests/distributed_test.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2025 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import sys
+import threading
+import unittest
+
+from absl.testing import absltest, parameterized
+
+import jax
+from jax._src import distributed
+from jax._src import test_util as jtu
+
+try:
+  import portpicker
+except ImportError:
+  portpicker = None
+
+jax.config.parse_flags_with_absl()
+
+@unittest.skipIf(not portpicker, "Test requires portpicker")
+class DistributedTest(jtu.JaxTestCase):
+  # TODO(phawkins): Enable after https://github.com/jax-ml/jax/issues/11222
+  # is fixed.
+  @unittest.SkipTest
+  def testInitializeAndShutdown(self):
+    if not jtu.test_device_matches(["gpu"]):
+      self.skipTest("Test only works with GPUs.")
+    # Tests the public APIs. Since they use global state, we cannot use
+    # concurrency to simulate multiple tasks.
+    port = portpicker.pick_unused_port()
+    jax.distributed.initialize(
+      coordinator_address=f"localhost:{port}", num_processes=1, process_id=0
+    )
+    jax.distributed.shutdown()
+
+  @parameterized.parameters([1, 2, 4])
+  def testConcurrentInitializeAndShutdown(self, n):
+    if not jtu.test_device_matches(["gpu"]):
+      self.skipTest("Test only works with GPUs.")
+    port = portpicker.pick_unused_port()
+
+    def task(i):
+      # We can't call the public APIs directly because they use global state.
+      state = distributed.State()
+      state.initialize(coordinator_address=f"localhost:{port}", num_processes=n, process_id=i)
+      state.shutdown()
+
+    threads = [threading.Thread(target=task, args=(i,)) for i in range(n)]
+    for thread in threads:
+      thread.start()
+    for thread in threads:
+      thread.join()
+
+  @jtu.sample_product(run_initialize=[True, False])
+  def test_is_distributed_initialized_multi_process(self, run_initialize):
+    port = portpicker.pick_unused_port()
+    pycmd = (
+      "import jax; "
+      + (
+        f"jax.distributed.initialize('localhost:{port}', 1, 0); "
+        if run_initialize
+        else ""
+      )
+      + 'print(jax.distributed.is_initialized(), end="")'
+    )
+    args = [sys.executable, "-c", pycmd]
+    process = subprocess.Popen(
+      args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True
+    )
+
+    try:
+      out, _ = process.communicate()
+      self.assertEqual(process.returncode, 0)
+      self.assertEqual(out, str(run_initialize))
+    finally:
+      process.kill()
```

Comment:
Is this necessary? Like when would the above try block error such that the process isn't already dead?

Reply:
Yeah probably not. I think subprocess can throw KeyboardInterrupt errors, but happy to roll without the try except block and see if it causes problems.

Reply:
Presumably this practice would be to make sure you kill the subprocess in the event calling `process.communicate()` throws.

Reply:
Switched to the more modern `subprocess.run`